features/cellular/framework/targets/TELIT/ME910/TELIT_ME910_CellularNetwork.cpp

```diff
@@ -42,4 +42,24 @@ nsapi_error_t TELIT_ME910_CellularNetwork::set_access_technology_impl(RadioAcces
     }
 }
 
+nsapi_error_t TELIT_ME910_CellularNetwork::do_user_authentication()
+{
+#if defined(MBED_CONF_NSAPI_DEFAULT_CELLULAR_USERNAME) && defined(MBED_CONF_NSAPI_DEFAULT_CELLULAR_PASSWORD)
+    if (!_device.get_property(AT_CellularDevice::PROPERTY_AT_CGAUTH)) {
+        return NSAPI_ERROR_UNSUPPORTED;
+    }
+    const bool stored_debug_state = _at.get_debug();
+    _at.set_debug(false);
+
+    _at.at_cmd_discard("+CGAUTH", "=", "%d%d%s%s", 1, CellularContext::AuthenticationType::CHAP, MBED_CONF_NSAPI_DEFAULT_CELLULAR_USERNAME, MBED_CONF_NSAPI_DEFAULT_CELLULAR_PASSWORD);
```

Comment:
Use `nsapi_error_t err = _at.at_cmd_discard()` here and then `err != NSAPI_ERROR_OK` later instead of `get_last_error()`.

Reply:
@AriParkkila Changes made; please review.

Thanks!